### ISSUE TITLE
Runs as root in Kubernetes only

### ISSUE THREAD

**Issue by kathoef** [martinclaus/image-registry-checker#1]
Deploying this application as a Kubernetes service basically works,

```
---
apiVersion: v1
kind: Namespace
metadata:
  name: registry-checker
---
apiVersion: v1
kind: Pod
metadata:
  name: registry-checker
  namespace: registry-checker
  labels:
    app.kubernetes.io/name: registry-checker
spec:
  nodeSelector:
    node.kubernetes.io/microk8s-controlplane: microk8s-controlplane
  containers:
    - name: registry-checker
      image: image-registry-checker-v2023.05.09
      ports:
        - containerPort: 80
 ---
apiVersion: v1
kind: Service
metadata:
  name: registry-nodeport-service
  namespace: registry-checker
spec:
  type: NodePort
  selector:
    app.kubernetes.io/name: registry-checker
  ports:
    - protocol: TCP
      port: 8080
      nodePort: 31385
      targetPort: 80
```

but not if a proper pod/container security context is applied,

```
---
apiVersion: v1
kind: Pod
metadata:
  name: registry-checker
  namespace: registry-checker
  labels:
    app.kubernetes.io/name: registry-checker
spec:
  nodeSelector:
    node.kubernetes.io/microk8s-controlplane: microk8s-controlplane
  securityContext:
    runAsUser: 1001
    runAsGroup: 1001
    runAsNonRoot: true
  containers:
    - name: registry-checker
      image: image-registry-checker-v2023.05.09
      ports:
        - containerPort: 80
          protocol: TCP
```

which causes the following startup error,

```
$ microk8s kubectl logs registry-checker -n registry-checker 
 2023-05-10T09:06:38.967Z INFO  image_registry_checker > Cannot read environment from .env: path not found
thread 'main' panicked at 'error binding to 0.0.0.0:80: error creating server listener: Permission denied (os error 13)', /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/warp-0.3.3/src/server.rs:213:27
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Would it be possible to come up with an implementation that runs as non root user as well? Otherwise I probably need to fiddle with Linux capabilities. 🤔 I am not familiar with Rust, but I think [the chosen socket approach](https://github.com/martinclaus/image-registry-checker/blob/963955e6263ed0bf063ccde0ca6fbcf88661ad07/image-registry-checker/src/main.rs#L32) is the problem.

**Comment by martinclaus** [comment]
I guess the problem is that the web server tries to bind to port `80`, which is not allowed by default for non-root user. That is surprising to me since the default port within the container should be `8080`.

**Comment by martinclaus** [comment]
And here is the culprit: https://github.com/martinclaus/image-registry-checker/blob/963955e6263ed0bf063ccde0ca6fbcf88661ad07/Dockerfile#L21

Adjusting the docker file should fix the problem. Sorry for the misleading README!

**Comment by martinclaus** [comment]
Fixed in #2

**Comment by kathoef** [comment]
Ah, of course... thanks! 😅
